### PR TITLE
client/http: introduce caller ID into the HTTP client

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #7300.

### What is changed and how does it work?

```commit-message
Introduce caller ID into the HTTP client.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by CabinfeverB — APPROVED** [review]

File: client/http/client.go [hunk 1]
```diff
@@ -196,13 +218,19 @@ func (c *client) execDuration(name string, duration time.Duration) {
 	c.executionDuration.WithLabelValues(name).Observe(duration.Seconds())
 }
 
+// Header key definition constants.
+const (
+	pdAllowFollowerHandleKey = "PD-Allow-Follower-Handle"
+	componentSignatureKey    = "component"
```
Comment (niubell) [inline client/http/client.go:224]: We can have a more intuitive name like 'CallerID', and it's a filed in http header, "X-Caller-ID" maybe a better name according HTTP RFC docs @JmPotato

File: client/http/client.go [hunk 2]
```diff
@@ -239,6 +267,8 @@ func (c *client) request(
 	logFields := []zap.Field{
 		zap.String("name", name),
 		zap.String("url", url),
+		zap.String("method", method),
+		zap.String("caller-id", c.callerID),
```
Comment (niubell) [inline client/http/client.go:271]: Do we log the caller id on the PD server side?

File: client/http/client.go (hunk 1 above)
Comment (JmPotato) [reply client/http/client.go:224]: Using `component` here is to reuse https://github.com/tikv/pd/blob/master/pkg/utils/requestutil/request_info.go#L52, but you're right about the naming, we could create a new filed to distinguish with it.

File: client/http/client.go (hunk 2 above)
Comment (JmPotato) [reply client/http/client.go:271]: It will appear in the audit log also.

**Pull request merged**